cds-snc/notification-api
# Add a 408 response code case

# Summary | Résumé

This PR does the following:
- add a 408 response code case, since we want to send that from dd-api when the scan times out
- added a new test for the "scan timed out" case
- target only python 3.10 in our black config so it will not fail on match statements (was more relevant for an earlier iteration of this PR, but though I'd keep it anyway)

To test:
- run this locally along with this branch of the dd-api: https://github.com/cds-snc/notification-document-download-api/pull/109
- modify the dd-api to return the 408 / "scan timed out" response by: a) adding`raise ScanInProgressError("Content scanning is in progress")` to https://github.com/cds-snc/notification-document-download-api/blob/main/app/utils/store.py#L112, and b) modifying [this timeout](https://github.com/cds-snc/notification-document-download-api/blob/main/app/download/views.py#L24) to be 60s: `SCAN_TIMEOUT_SECONDS = 1 * 60`)
- verify that the email and attachment are sent as expected after about 1 minute when the timeout is reached.
